Rename GraphObject & GraphList that's backwards-compatible

SammyK opened the pull request:
This will close #320.

Now that we're adopting SemVer we can use `MAJOR` versions to deprecate stuffs. Yay!

It looks like a big PR, but it's really just a tiny rename that touches many places in the codebase.

- `GraphObject` is now `GraphNode` & `GraphObject` is deprecated with the `@deprecated` flag.
- `GraphList` is now `GraphEdge` & `GraphList` is deprecated with the `@deprecated` flag.
- `GraphObjectFactory` is now `GraphNodeFactory` & `GraphObjectFactory` is deprecated with the `@deprecated` flag.
- `FacebookResponse::getGraphList()` has been `@deprecated` & renamed to `getGraphEdge()`
- `FacebookResponse::getGraphObject()` has been `@deprecated` & renamed to `getGraphNode()`

This shouldn't be a breaking change for anyone on 4.1 unless they've really overwritten deep functionality of the codebase.

I've still got to update the docs so don't merge yet. :)

SammyK commented:
Docs updated!

Just realized that this is going to hardcore break @devmsh's PR #405, so you might want to merge his in first when he finalizes his changes. Then I'll go in an fix any conflicts on this PR before merging. :)

SammyK commented:
Fixed conflicts with #405 so this should be good to go pending any feedback from @yguedidi :)

yguedidi commented on docs/FacebookRequest.fbmd:
```diff
@@ -51,9 +51,9 @@ try {
 }
 
-$object = $response->getGraphObject();
+$node = $response->getGraphNode();
```
Why not `$graphNode` here too?

yguedidi commented on docs/GraphNode.fbmd:
```diff
@@ -25,8 +25,8 @@ $fb = new Facebook\Facebook(\* *\);
-// Get the base class GraphObject from the response
-$node = $response->getGraphObject();
+// Get the base class GraphNode from the response
+$node = $response->getGraphNode();
```
`$graphNode` herer too

yguedidi commented on docs/example_pagination_basic.fbmd:
```diff
@@ -32,7 +32,7 @@ try {
 
 // Page 1
-$feed = $response->getGraphList();
+$feed = $response->getGraphEdge();
```
`$feedEdge`?

yguedidi commented on tests/GraphNodes/GraphObjectFactoryTest.php:
```diff
@@ -235,203 +73,42 @@ public function testAGraphObjectWillBeCastAsAGraphObject()
         $factory = new GraphObjectFactory($res);
-        $graphObject = $factory->makeGraphObject();
-        $graphData = $graphObject->asArray();
+        $graphEdge = $factory->makeGraphList();
```
`makeGraphEdge`?

SammyK replied:
This is to show BC but renamed `$graphEdge` to `$graphList` to be more explicit what it's testing. :)

yguedidi commented on tests/GraphNodes/GraphObjectFactoryTest.php:
```diff
@@ -235,203 +73,42 @@ public function testAGraphObjectWillBeCastAsAGraphObject()
+        $graphData = $graphEdge->asArray();
 
-        $this->assertInstanceOf('\Facebook\GraphNodes\GraphObject', $graphObject);
+        $this->assertInstanceOf('\Facebook\GraphNodes\GraphList', $graphEdge);
```
`'\Facebook\GraphNodes\GraphEdge'`?

SammyK replied:
Also to show BC but renamed `$graphEdge` to `$graphList`.

SammyK commented:
Thanks for the review @yguedidi! Done. :)

SammyK commented:
Also just updated per [@yguedidi's suggestion](https://github.com/facebook/facebook-php-sdk-v4/issues/429#issuecomment-101688077). Renamed `getProperty()` to `getField()` and deprecated `getProperty()`.

gfosco commented:
![Awesome](http://i.giphy.com/7eAvzJ0SBBzHy.gif)

SammyK commented:
![](http://media.giphy.com/media/q5iwtmYTOjs7S/giphy.gif)
